TrackingSystem/src/main/java/de/telran/controller/CustomerController.java

```diff
@@ -1,33 +1,79 @@
 package de.telran.controller;
 
+import de.telran.dto.CustomerDTO;
 import de.telran.entity.Customer;
+import de.telran.entity.Shipment;
 import de.telran.service.CustomerService;
+import org.modelmapper.ModelMapper;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.PostMapping;
-import org.springframework.web.bind.annotation.RequestBody;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.*;
 
 import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
 
 @RestController
 public class CustomerController {
 
     CustomerService customerService;
 
+    ModelMapper modelMapper = new ModelMapper();
+
     @Autowired
     public CustomerController(CustomerService customerService) {
         this.customerService = customerService;
     }
 
     @GetMapping("/api/customers")
-    List<Customer> getAllCustomers() {
-        return customerService.getAllCustomers();
+    List<CustomerDTO> getAllCustomers() {
+        return  customerService.getAllCustomers()
+                .stream()
+                .map(this::convertCustomerToCustomerDTO)
+                .collect(Collectors.toList());
+    }
+
+    @GetMapping("api/customers/{customer_id}/shipments")
+    CustomerDTO getAllShipmentsOfACustomer(@PathVariable("customer_id") Long id) {
+        Customer customer = customerService.getCustomerById(id);
+        return convertCustomerToCustomerDTO(customer);
     }
 
+    @GetMapping("api/customers/{customer_id}/shipments/{shipment_id}/statuses")
+    Optional<Shipment> getCurrentStatusesOfAShipmentOfACustomer(@PathVariable("customer_id") Long customerId,
+                                                                @PathVariable("shipment_id") Long shipmentId) {
```

Comment:
But here is the same issue as above: your URL returns a collection of statuses, but you return a shipment. For this case I would just  define the URL like this: `/api/customers/{customer_id}/shipments/{shipment_id}`, which means return a shipment by id with all its statuses (and probably customers). This is similar to what we discussed last time when a Shipment has "one to many" as a `List<Status>`.